lua/tests/infoview_lsp_spec.lua

```diff
@@ -0,0 +1,62 @@
+local infoview = require('lean.infoview')
+local helpers = require('tests.helpers')
+
+local function get_info_lines(infoview_info)
+  return table.concat(vim.api.nvim_buf_get_lines(infoview_info.bufnr, 0, -1, true), "\n")
+end
+
+local function infoview_lsp_test(pos, expected)
+    local infoview_info = infoview.open()
+    vim.api.nvim_win_set_cursor(0, pos)
+    infoview.update(infoview_info.bufnr)
+    local result, _ = vim.wait(5000, function()
+      for _, string in pairs(expected) do
+        local curr = get_info_lines(infoview_info)
+        if not curr:find(string) then return false end
```

Comment:
It seems potentially a bit surprising that this does a partial match. Is it much more annoying to just return the contents after the wait, and let the caller use whichever assertion they want? And if luassert doesn't already have one (ugh) we add `assert.contains(foo, bar)`? Otherwise I guess fine as is, but yeah seems easy to forget this is not matching exactly.